transport/src/main/java/io/netty/channel/AbstractChannelHandlerContext.java

```diff
@@ -977,19 +1123,19 @@ private boolean isNotValidPromise(ChannelPromise promise, boolean allowVoidPromi
         return false;
     }
 
-    private AbstractChannelHandlerContext findContextInbound() {
+    private AbstractChannelHandlerContext findContextInbound(int mask) {
         AbstractChannelHandlerContext ctx = this;
         do {
             ctx = ctx.next;
-        } while (!ctx.inbound);
+        } while ((ctx.executionMask & mask) == 0);
         return ctx;
     }
 
-    private AbstractChannelHandlerContext findContextOutbound() {
+    private AbstractChannelHandlerContext findContextOutbound(int mask) {
         AbstractChannelHandlerContext ctx = this;
         do {
             ctx = ctx.prev;
-        } while (!ctx.outbound);
+        } while ((ctx.executionMask & mask) == 0);
```

Comment:
Test that checks that handlers are actually skipped with `@Skip` and not skipped without it?

Reply:
good idea..

Reply:
@doom369 done